extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -69,30 +68,36 @@ class AmpAccordion extends AMP.BaseElement {
           'Each section must have exactly two children. ' +
           'See https://github.com/ampproject/amphtml/blob/master/extensions/' +
           'amp-accordion/amp-accordion.md. Found in: %s', this.element);
-
       this.mutateElement(() => {
-        const header = sectionComponents_[0];
         const content = sectionComponents_[1];
-        header.classList.add('i-amphtml-accordion-header');
-        header.setAttribute('role', 'tab');
         content.classList.add('i-amphtml-accordion-content');
-        content.setAttribute('role', 'tabpanel');
         let contentId = content.getAttribute('id');
         if (!contentId) {
           contentId = this.element.id + '_AMP_content_' + index;
           content.setAttribute('id', contentId);
         }
+
         if (this.currentState_[contentId]) {
           section.setAttribute('expanded', '');
         } else if (this.currentState_[contentId] == false) {
           section.removeAttribute('expanded');
         }
+
+        const header = sectionComponents_[0];
+        header.classList.add('i-amphtml-accordion-header');
+        header.setAttribute('role', 'heading');
         header.setAttribute('aria-controls', contentId);
         header.setAttribute('aria-expanded',
             section.hasAttribute('expanded').toString());
-        header.addEventListener('click', this.onHeaderClick_.bind(this));
+        if (!header.hasAttribute('tabindex')) {
+          header.setAttribute('tabindex', 0);
+        }
+        this.headers_.push(header);
       });
     });
+
+    this.element.addEventListener('click', this.clickHandler_.bind(this));
```

Comment:
let's install there on the `header` instead, looping through headers to find the target for every click inside the element is unnecessary.